Improve instructions for db-file fixtures

fblanchetNaN opened the pull request:
<!--

Thanks for submitting a pull request against QCoDeS.

To help us effectively merge your pr please consider the following check list.

- [ ] Make sure that the pull request contains a short description of the changes made.
- [ ] If you are submitting a new feature please document it. This can be in the form of inline
      docstrings, an example notebook or restructured text files.
- [ ] Please include automatic tests for the changes made when possible.

Unless your change is a small or trivial fix please add a small changelog entry:

- [ ] Create a file in the docs\changes\newsfragments folder with a short description of the change.

This file should be in the format number.categoryofcontribution. Here the number should either be the number
of the pull request. To get the number of the pull request one must
first the pull request and then subsequently update the number. The category of contribution should be
one of ``breaking``, ``new``, ``improved``, ``new_driver`` ``improved_driver``, ``underthehood``.

If this fixes a known bug reported against QCoDeS:

- [ ] Please include a string in the following form ``closes #xxx`` where ``xxx``` is the number of the bug fixed.

Please have a look at [the contributing guide](https://qcodes.github.io/Qcodes/community/contributing.html)
for more information.

If you are in doubt about any of this please ask and we will be happy to help.

-->

Improve `pytest.skip` messages when db-file are missing to point out at submodule qcodes_db_fixtures.

Explicitly recommend cloning with submodules, include the `git clone` command and options in the documentation.

Fix related error 404 in CONTRIBUTING.rst.

See QCoDeS/qcodes_generate_test_db#8

jenshnielsen commented:
@fblanchetNaN Sorry it seemed like our prs have crossed. I am happy to merge yours, but I think it would be nice to do the refactor from mine to avoid duplication of the skip message

fblanchetNaN commented:
> @fblanchetNaN Sorry it seemed like our prs have crossed. I am happy to merge yours, but I think it would be nice to do the refactor from mine to avoid duplication of the skip message

I rebased on your commit, adjusted my code to yours, and force-pushed my modifications.

jenshnielsen commented on CONTRIBUTING.rst:
```diff
@@ -76,7 +76,7 @@ Setup
 ~~~~~
 
 -  Clone and register the package for development as described in the
-   `README <README.md#installation>`__
+   `here <http://qcodes.github.io/Qcodes/start/index.html#installation>`__
```
here does not seem to match the line above.

fblanchetNaN replied:
I removed "in the" and keep "here" for the link to be coherent with the style of README.rst.

jenshnielsen reviewed (approved):
Bors merge

jenshnielsen commented:
bors merge
